core/node/base_token_fetcher/src/lib.rs

```diff
@@ -1,19 +1,24 @@
-use std::{cmp::min, sync::Arc, time::Duration};
-
-use anyhow::Context;
+use std::{
+    cmp::min,
+    str::FromStr,
+    sync::{Arc, Mutex as StdMutex},
+    time::Duration,
+};
+
+use anyhow::{anyhow, Context};
 use async_trait::async_trait;
 use hex::ToHex;
-use metrics::atomics::AtomicU64;
 use tokio::{sync::Mutex, time::sleep};
 use zksync_config::configs::BaseTokenFetcherConfig;
+use zksync_dal::BigDecimal;
```

Comment:
I don't think it's adequate to use `zksync_dal` just to get `BigDecimal` from it. Please use the correct dependency (e.g. one that `zksync_dal` (maybe transitively) uses).

Reply:
Correct, fixed it.